From: Ralf Baechle <ralf@linux-mips.org>
Date: Sat, 9 Dec 2006 16:12:18 +0000 (+0000)
Subject: [MIPS] Fix build of several IDE drivers by providing pci_get_legacy_ide_irq
X-Git-Tag: linux-2.6.20-rc1^0~18
X-Git-Url: http://www.linux-mips.org/git?p=linux.git;a=commitdiff_plain;h=49a3ea926ff0cc6c5af15d79bfbab075991f5dab;hp=87816c5358a6dde9de43ae58bd06737095ec1a56

[MIPS] Fix build of several IDE drivers by providing pci_get_legacy_ide_irq

Signed-off-by: Ralf Baechle <ralf@linux-mips.org>
---

diff --git a/include/asm-mips/pci.h b/include/asm-mips/pci.h
index c4d68be..7f0f120 100644
--- a/include/asm-mips/pci.h
+++ b/include/asm-mips/pci.h
@@ -187,4 +187,10 @@ static inline void pcibios_add_platform_entries(struct pci_dev *dev)
 /* Do platform specific device initialization at pci_enable_device() time */
 extern int pcibios_plat_dev_init(struct pci_dev *dev);
 
+/* Chances are this interrupt is wired PC-style ...  */
+static inline int pci_get_legacy_ide_irq(struct pci_dev *dev, int channel)
+{
+	return channel ? 15 : 14;
+}
+
 #endif /* _ASM_PCI_H */
